### DISCUSSION TITLE
I wanted to know where bun keeps cache of imported modules, node js keeps it in require.cache object?

### DISCUSSION THREAD

**Discussion by tsp1998** [oven-sh/bun#10162] [Q&A] [answered]
`
const main = async () => {
  console.log(require.cache)
  // const parse = customRequire('dotenv', 'parse')
  const parse = require('dotenv').parse
  // const { parse } = await import('dotenv')
  console.log(require.cache)
  console.log(`parse`, parse)
}
main()
`

![image](https://github.com/oven-sh/bun/assets/43929859/409986a9-1415-44b3-9354-48bcb89b3477)

-------------------

I ran this code with node and bun, in node it shows all imported modules in cache but with bun it shows empty object for cache.

**Comment by guest271314** [comment]
On Linux `~/.bun/install/cache`. Try runnin `bun pm --help`.

**Reply by tsp1998** [reply]
Thank you but I mean runtime cache not packages cache

**Reply by tsp1998** [reply]
I mean I need to check which modules imported in memory cache at runtime

**Comment by Jarred-Sumner** [comment, marked as answer]
You can use `require.cache`

The empty `require.cache` is a bug in our `console.log` implementation

**Comment by sculpt0r** [comment]
Sorry for awaking this topic, but does it work with `import`? Nodejs says it is not and imports cache is `unexposed`, but how about Bun?

I'm trying to invalidate cache for certain modules that are loaded via `import ... from`. Is it achievable simple by doing `delete require.cache[moduleName]`?

**Reply by Jarred-Sumner** [reply]
yes it works with import too

**Reply by gherciu** [reply]
@Jarred-Sumner Seems that no
If you try to require a file that uses inside `import` it just says `Canot use import outside a module`
I mean only local files I tried no external deps at all

**Comment by gherciu** [comment]
@Jarred-Sumner 
Seems like require.cache is always empty, and is not just a console.log bug
When I try to access somethign inside it always returns undefined even if the path should be there since was required in code
